### PR TITLE
Check allowance of SendLocation by "deliveryMode" value

### PR DESCRIPTION
**Pull request by anosach-luxoft**
1. In case
mobile app sends the SendLocation_request with “deliveryMode” parameter and other params 
and “deliveryMode” parameter is allowed by Policies 
SDL must: 
transfer _SendLocation_request to HMI
respond with <resultCode_received_from_HMI> to mobile app
 	 	 
2. In case
mobile app sends the SendLocation_request with “deliveryMode” parameter and other params 
and “deliveryMode” parameter is NOT allowed by Policies 
SDL must: 
cut off the "deliveryMode" parameter from SendLocation request
transfer SendLocation without "deliveryMode" parameter to HMI
respond with <resultCode_received_from_HMI> to mobile app

See [full](https://adc.luxoft.com/jira/browse/APPLINK-21103) CRQ description.
[JIRA](https://adc.luxoft.com/jira/browse/APPLINK-23313) ticket.

@dev-gh @AGaliuzov please review

### CODE REVIEW

**Comment by dev-gh** [comment]
Reviewed

File: src/components/application_manager/src/commands/mobile/send_location_request.cc [hunk 1]
```diff
@@ -60,9 +62,17 @@ void SendLocationRequest::Run() {
...
+  if (msg_params.keyExists(strings::delivery_mode)) {
+    if (parameters_permissions().allowed_params.end() ==
+        std::find(parameters_permissions().allowed_params.begin(),
+                  parameters_permissions().allowed_params.end(),
+                  strings::delivery_mode)) {
+      msg_params.erase(strings::delivery_mode);
+    }
+  }
 
+  std::list<Common_TextFieldName::eType> fields_to_check;
```
Comment (AGaliuzov) [inline src/components/application_manager/src/commands/mobile/send_location_request.cc]: please do not use the `list` use `vector` instead

**Comment by AGaliuzov** [comment]
Reviewed

File: src/components/interfaces/HMI_API.xml [hunk 2]
```diff
@@ -1182,6 +1182,13 @@
   </element>
 </enum>
 
+<enum name="DeliveryMode">
```
Comment (LuxoftAKutsan) [inline src/components/interfaces/HMI_API.xml]: @anosach-luxoft  As I know if we change HMI API we should increase it's version and update date in HEAD.

File: src/components/interfaces/MOBILE_API.xml [hunk 3]
```diff
@@ -797,6 +797,13 @@
     <element name="DYNAMIC" />
   </enum>
 
+  <enum name="DeliveryMode">
```
Comment (LuxoftAKutsan) [inline src/components/interfaces/MOBILE_API.xml]: @anosach-luxoft  As I know if we change HMI API we should increase it's minor version version.

File: src/components/interfaces/HMI_API.xml [hunk 4]
```diff
@@ -3178,6 +3178,9 @@
     <param name="locationImage" type="Common.Image" mandatory="false">
       <description>     Image / icon of intended location (if applicable and supported)   </description>
     </param>
+    <param name="deliveryMode" type="Common.DeliveryMode" mandatory="false">
```
Comment (LuxoftAKutsan) [inline src/components/interfaces/HMI_API.xml]: @anosach-luxoft  Please squash commits with API changes if API changes don't affect SDL build and work. If it affects, please squash API and logic commits

File: src/components/application_manager/src/commands/mobile/send_location_request.cc [hunk 5]
```diff
@@ -60,9 +62,17 @@ void SendLocationRequest::Run() {
     return;
   }
 
-  const smart_objects::SmartObject& msg_params = (*message_)[strings::msg_params];
-  std::list<Common_TextFieldName::eType> fields_to_check;
+  smart_objects::SmartObject& msg_params = (*message_)[strings::msg_params];
+  if (msg_params.keyExists(strings::delivery_mode)) {
+    if (parameters_permissions().allowed_params.end() ==
+        std::find(parameters_permissions().allowed_params.begin(),
+                  parameters_permissions().allowed_params.end(),
```
Comment (Kozoriz) [inline src/components/application_manager/src/commands/mobile/send_location_request.cc]: @anosach-luxoft , if we do some action more than one time, we can move result of action to temporary link.

**Comment by LuxoftAKutsan** [comment]
reviewed. But Please squash commits required before merge

**Comment by anosach-luxoft** [comment]
@AGaliuzov @LuxoftAKutsan @Kozoriz PR has been rebased and updated.
@LuxoftAKutsan question with API version increment is still open so lets wait for results

File: src/components/application_manager/src/commands/mobile/send_location_request.cc (hunk 5 above)
Comment (Kozoriz) [inline src/components/application_manager/src/commands/mobile/send_location_request.cc]: @anosach-luxoft , are you agree with me, or no?

**Comment by TMelnyk** [comment]
MOBILE_API and HMI_API reviewed

**Pull request merged**